src/Tizen.NUI/src/public/Events/PanGestureDetector.cs

```diff
@@ -71,19 +71,25 @@ internal PanGestureDetector(global::System.IntPtr cPtr, bool cMemoryOwn) : base(
                 if (detectedEventHandler == null)
                 {
                     detectedCallback = OnPanGestureDetected;
-                    DetectedSignal().Connect(detectedCallback);
+                    using PanGestureDetectedSignal signal = new PanGestureDetectedSignal(Interop.PanGestureDetector.DetectedSignal(SwigCPtr), false);
+                    signal?.Connect(detectedCallback);
                 }
-
                 detectedEventHandler += value;
             }
-
             remove
             {
                 detectedEventHandler -= value;
-
-                if (detectedEventHandler == null && DetectedSignal().Empty() == false)
+                if (detectedEventHandler == null)
                 {
-                    DetectedSignal().Disconnect(detectedCallback);
+                    using PanGestureDetectedSignal signal = new PanGestureDetectedSignal(Interop.PanGestureDetector.DetectedSignal(SwigCPtr), false);
+                    if (signal?.Empty() == false)
```

Comment:
Just my curiosity :
From now on, should we check `Empty()` before calling `Disconnect` and null check?
AFAIK, we normally call Disconnect and set null without checking `Empty()` of signal.

Reply:
yes, you are right, it is not very necessary. but it is secure to check again the native dali signal is disconnected surely.
and also in Dispose(), the callback null check is performed so before Dispose(), setting null to callback should be done carefully. (because if the signal is not disconnected here and callback is set to null, then signal disconnection in Dispose() will be missed. this is why the double checking has been added.

Reply:
I see. Now this commit can be a leading code for double checking. Also, I can refer to it later~
Thank you for checking them!